maintainers/scripts/update.py

```diff
@@ -12,80 +14,150 @@
 
 class CalledProcessError(Exception):
     process: asyncio.subprocess.Process
+    stderr: Optional[bytes]
 
 class UpdateFailedException(Exception):
     pass
 
-def eprint(*args, **kwargs):
-    print(*args, file=sys.stderr, **kwargs)
-
-async def check_subprocess(*args, **kwargs):
+T = TypeVar('T')
+
+def make_object_from_json_value(value: Any, t: type[T]) -> T:
+    return cast(T, jsons.load(value, t))
+
+def eprint(*args: Any) -> None:
+    print(*args, file=sys.stderr)
+
+class Feature(Enum):
+    COMMIT = 'commit'
+
+@dataclass
+class PackageInfo:
+    name: str
+    pname: str
+    updateScript: list[str]
+    supportedFeatures: set[Feature]
+    oldVersion: str
+    attrPath: str
+
+@dataclass
+class ChangeInfo:
+    attrPath: str
+    oldVersion: str
+    newVersion: str
+    commitMessage: Optional[str]
+    commitBody: Optional[str]
+    files: list[str]
+
+@overload
+async def check_subprocess_output(
+    cmd: list[str],
+    capture_stdout: Literal[False] = False,
+    capture_stderr: bool = False,
+    cwd: Optional[str] = None,
+) -> None:
+    ...
+
+@overload
+async def check_subprocess_output(
+    cmd: list[str],
+    capture_stdout: Literal[True] = True,
+    capture_stderr: bool = False,
+    cwd: Optional[str] = None,
+) -> bytes:
+    ...
+
+async def check_subprocess_output(
+    cmd: list[str],
+    capture_stdout: bool = False,
+    capture_stderr: bool = False,
+    cwd: Optional[str] = None,
+) -> Optional[bytes]:
```

Comment:
This is too verbose for my taste. Maybe we should just have two different functions.